biz.aQute.bndlib/src/aQute/bnd/osgi/Verifier.java

```diff
@@ -531,6 +532,22 @@ public void verify() throws Exception {
 		verifyCapabilities();
 		verifyMetaPersistence();
 		verifyPathNames();
+
+		doVerifierPlugins();
+	}
+
+	private void doVerifierPlugins() {
+		for (VerifierPlugin plugin : getPlugins(VerifierPlugin.class)) {
+			try {
+				Processor previous = beginHandleErrors(plugin.toString());
+				plugin.verify(analyzer);
+				endHandleErrors(previous);
```

Comment:
Perhaps the endHandleErrors should be in a finally block since plugin.verify can throw an exception meaning endHandleErrors will not be called for that plugin. Analyzer.doPlugins (which you copied this loop from) has this same flaw.

```java
Processor previous = beginHandleErrors(plugin.toString());
try {
   plugin.verify(analyzer);
} finally {
   endHandleErrors(previous);
}
```